trie/transition.go

```diff
@@ -63,7 +66,23 @@ func (t *TransitionTrie) TryGet(addr, key []byte) ([]byte, error) {
 		return val, nil
 	}
 	// TODO also insert value into overlay
-	return t.base.TryGet(nil, key)
+	rlpval, err := t.base.TryGet(nil, key)
+	if err != nil {
+		return nil, err
+	}
+	if len(rlpval) == 0 {
+		return nil, nil
+	}
+	// the value will come as RLP, decode it so that the
+	// interface is consistent.
+	_, content, _, err := rlp.Split(rlpval)
+	if err != nil || len(content) == 0 {
+		fmt.Println(err)
+		return nil, err
```

Comment:
Maybe we should split the case when `len(content) == 0`?
In that case, printing and returning `err` would only print/return `nil` which can be confusing.

So maybe if `len(content) == 0` do some `errors.New(...)`.

Reply:
that was a debug thing, so I removed it. Do you still think that I should split both cases?

Reply:
>  Do you still think that I should split both cases?

No, actually I think this is fine. I thought the intention was to return a non-nil error if `len(content)==0`, but that's not actually the case.